util/cosmos/iscn/sign.js

```diff
@@ -125,6 +125,78 @@ function formatISCNPayload(payload, version = 1) {
   };
 }
 
+export async function estimateISCNTxFee(tx, {
+  version = 1,
+} = {}) {
+  const record = formatISCNPayload(tx);
+  const feePerByte = await queryFeePerByte();
+  const feePerByteAmount = feePerByte ? parseInt(feePerByte.amount, 10) : 1;
+  const {
+    recordNotes,
+    contentFingerprints,
+    stakeholders,
+    contentMetadata,
+  } = record;
+  const now = new Date();
+  const obj = {
+    '@context': {
+      '@vocab': 'http://iscn.io/',
+      recordParentIPLD: {
+        '@container': '@index',
+      },
+      stakeholders: {
+        '@context': {
+          '@vocab': 'http://schema.org/',
+          entity: 'http://iscn.io/entity',
+          rewardProportion: 'http://iscn.io/rewardProportion',
+          contributionType: 'http://iscn.io/contributionType',
+          footprint: 'http://iscn.io/footprint',
+        },
+      },
+      contentMetadata: {
+        '@context': null,
+      },
+    },
+    '@type': 'Record',
+    '@id': `iscn://${ISCN_REGISTRY_NAME}/btC7CJvMm4WLj9Tau9LAPTfGK7sfymTJW7ORcFdruCU/1`,
+    recordTimestamp: now.toISOString(),
+    recordVersion: version,
+    recordNotes,
+    contentFingerprints,
+    recordParentIPLD: {},
+  };
+  if (version > 1) {
+    obj.recordParentIPLD = {
+      '/': 'bahuaierav3bfvm4ytx7gvn4yqeu4piiocuvtvdpyyb5f6moxniwemae4tjyq',
+    };
+  }
+
+  const byteSize = Buffer.from(jsonStringify(obj), 'utf-8').length
+    + stakeholders.reduce((acc, s) => acc + s.length, 0)
+    + contentMetadata.length;
+  const feeAmount = byteSize * feePerByteAmount;
+
+  return feeAmount;
+}
+
+export async function estimateISCNTxGas(tx, version) {
+  const ISCN_FEE = await estimateISCNTxFee(tx, { version });
+  return {
+    gasFee: {
+      amount: [{ amount: (DEFAULT_GAS_PRICE_NUMBER * ISCN_GAS).toFixed(), denom: 'nanolike' }],
+      gas: ISCN_GAS.toFixed(),
+    },
+    iscnFee: ISCN_FEE,
+  };
+}
+
```

Comment:
Would separating `estimateISCNTxFee()` and `estimateISCNTxGas()` be clearer? 
Remove `iscnFee` from `estimateISCNTxGas`, and just call two functions if we want total
```
const gasFee = await estimateISCNTxGas(tx);
const iscnFee = await estimateISCNTxFee();
```
when using

Reply:
Yes, good idea :)